### ISSUE TITLE
"Given calling package android does not match caller's uid 2000" while using scrcpy on Xiaomi hyperOS2, the screen mirroring works normally but audio mirroring doesn't work

### ISSUE THREAD

**Issue by TBCJames** [Genymobile/scrcpy#5608]
_Please read the [prerequisites] to run scrcpy._

[prerequisites]: https://github.com/Genymobile/scrcpy#prerequisites

_Also read the [FAQ] and check if your [issue][issues] already exists._

[FAQ]: https://github.com/Genymobile/scrcpy/blob/master/FAQ.md
[issues]: https://github.com/Genymobile/scrcpy/issues

## Environment

 - **OS: macOS15.1, Windows11 26100**
 - **Scrcpy version: 2.7** [e.g. 2.5]
 - **Installation method: brew(macOS), Windows release(Windows11)** [e.g. manual build, apt, snap, brew, Windows release...]
 - **Device model: Xiaomi 14**
 - **Android version: HyperOS2 based on Android 15** [e.g. 14]

## Describe the bug

The screen mirroring works normally but audio mirroring doesn't work, sound still comes out of the device instead of my PC.

```
[server] INFO: Device: [Xiaomi] Xiaomi 23127PN0CC (Android 15)
[server] ERROR: Exception on thread Thread[audio-encoder,5,main]
java.lang.SecurityException: Given calling package android does not match caller's uid 2000
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3256)
	at android.os.Parcel.createException(Parcel.java:3240)
	at android.os.Parcel.readException(Parcel.java:3223)
	at android.os.Parcel.readException(Parcel.java:3165)
	at android.app.IActivityManager$Stub$Proxy.getContentProvider(IActivityManager.java:6695)
	at android.app.ActivityThread.acquireProvider(ActivityThread.java:8666)
	at android.app.ContextImpl$ApplicationContentResolver.acquireProvider(ContextImpl.java:3804)
	at android.content.ContentResolver.acquireProvider(ContentResolver.java:2529)
	at android.provider.Settings$ContentProviderHolder.getProvider(Settings.java:3317)
	at android.provider.Settings$NameValueCache.getStringsForPrefixStripPrefix(Settings.java:3760)
	at android.provider.Settings$NameValueCache.-$$Nest$mgetStringsForPrefixStripPrefix(Unknown Source:0)
	at android.provider.Settings$Config.getStrings(Settings.java:20437)
	at android.provider.Settings$Config.getStrings(Settings.java:20388)
	at android.provider.SettingsConfigDataStore.getProperties(SettingsConfigDataStore.java:48)
	at android.provider.DeviceConfig.getPropertiesWithoutOverrides(DeviceConfig.java:1125)
	at android.provider.DeviceConfig.getProperties(DeviceConfig.java:1115)
	at android.media.codec.FeatureFlagsImpl.load_overrides_codec_fwk(FeatureFlagsImpl.java:19)
	at android.media.codec.FeatureFlagsImpl.setCallbackStall(FeatureFlagsImpl.java:146)
	at android.media.codec.Flags.setCallbackStall(Flags.java:109)
	at android.media.MediaCodec.setCallback(MediaCodec.java:5280)
	at com.genymobile.scrcpy.FakeContext$$ExternalSyntheticApiModelOutline0.m(D8$$SyntheticClass:0)
	at com.genymobile.scrcpy.audio.AudioEncoder.encode(AudioEncoder.java:199)
	at com.genymobile.scrcpy.audio.AudioEncoder.lambda$start$0$com-genymobile-scrcpy-audio-AudioEncoder(AudioEncoder.java:131)
	at com.genymobile.scrcpy.audio.AudioEncoder$$ExternalSyntheticLambda3.run(D8$$SyntheticClass:0)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.am.ContentProviderHelper.getContentProvider(ContentProviderHelper.java:147)
	at com.android.server.am.ActivityManagerService.getContentProvider(ActivityManagerService.java:7710)
	at android.app.IActivityManager$Stub.onTransact$getContentProvider$(IActivityManager.java:13102)
	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3429)
	at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:3027)

INFO: Renderer: metal
WARN: Demuxer 'audio': stream explicitly disabled by the device
INFO: Texture: 1200x2664
```

**Comment by TBCJames** [comment]
Has been fixed at version 3.0.2

**Comment by rom1v** [comment]
Yes, it was fixed in 3.0 by #5476.
